src/manifest/mod.rs

```diff
@@ -210,8 +211,25 @@ impl CrateData {
         }
         let manifest = fs::read_to_string(&manifest_path)
             .with_context(|_| format!("failed to read: {}", manifest_path.display()))?;
-        let manifest: CargoManifest = toml::from_str(&manifest)
-            .with_context(|_| format!("failed to parse manifest: {}", manifest_path.display()))?;
+        let manifest = &mut toml::Deserializer::new(&manifest);
+
+        let mut unused_keys = BTreeSet::new();
+
+        let manifest: CargoManifest = serde_ignored::deserialize(manifest, |path| {
+            let path_string = path.to_string();
+
+            if path_string.contains("metadata") {
```

Comment:
This may want to be `metadata.wasm-pack` perhaps? That way we won't accidentally warn about other `metadata` sections accidentally

Reply:
Oh right! That makes sense. Thanks!

Reply:
@alexcrichton: Hmm, if I do `metadata.wasm-pack` we wouldn't warn about misspelled `wasm-pack`. i.e. if someone writes `[package.metadata.wasmpack.profile.dev.wasm-bindgen]`.

Reply:
That's true, but I personally would think that we can't land this while it warns about `[package.metadata.some-unrelated-tool]`. 

We can possibly do a lehvenstein distance thing (typically used for "did you typo this?") to figure out if it looks like `wasm-pack` specifically, but other crates/tools using `[package.metadata]` should not be warned about when using `wasm-pack`

Reply:
Sounds reasonable! Pushed some code for this.